jobbergate-agent/jobbergate_agent/identity/cluster_api.py

```diff
@@ -0,0 +1,128 @@
+"""Core module for Jobbergate API identity management"""
+import typing
+
+import httpx
+import jwt
+
+from jobbergate_agent.settings import SETTINGS
+from jobbergate_agent.utils.exception import AuthTokenError
+from jobbergate_agent.utils.logging import logger
+
+
+CACHE_DIR = SETTINGS.CACHE_DIR / "cluster-api"
+
+
+def _load_token_from_cache() -> typing.Union[str, None]:
+    """
+    Looks for and returns a token from a cache file (if it exists).
+    Returns None if::
+    * The token does not exist
+    * Can't read the token
+    * The token is expired (or will expire within 10 seconds)
+    """
+    token_path = CACHE_DIR / "token"
+    if not token_path.exists():
+        return None
+
+    try:
+        token = token_path.read_text()
+    except Exception:
+        logger.warning(f"Couldn't load token from cache file {token_path}. Will acquire a new one")
+        return None
+
+    try:
+        jwt.decode(token, options=dict(verify_signature=False, verify_exp=True), leeway=-10)
+    except jwt.ExpiredSignatureError:
+        logger.warning("Cached token is expired. Will acquire a new one.")
+        return None
+
+    return token
+
+
+def _write_token_to_cache(token: str):
+    """
+    Writes the token to the cache.
+    """
+    if not CACHE_DIR.exists():
+        logger.debug("Attempting to create missing cache directory")
+        try:
+            CACHE_DIR.mkdir(mode=0o700, parents=True, exist_ok=True)
+        except Exception:
+            logger.warning(f"Couldn't create missing cache directory {CACHE_DIR}. Token will not be saved.")  # noqa
+            return
+
+    token_path = CACHE_DIR / "token"
+    try:
+        token_path.write_text(token)
+    except Exception:
+        logger.warning(f"Couldn't save token to {token_path}")
+
+
+def acquire_token() -> str:
```

Comment:
We need some logic the reacquire them if necessary. Now that the agent runs continually, they eventually will expire.